### ISSUE TITLE
Patching an import module results raises no-value-for-parameter incorrectly

### ISSUE THREAD

**Issue by xNinjaKittyx** [PyCQA/pylint#5963]
### Bug description

Here's an `example.py` . It's a bit silly, but works.

```python
import requests

def new_get(uri, *args, **kwargs):
    print(uri, args, kwargs)


requests.get = new_get


requests.get(uri="https://google.com")
```

The error in concern
```
phantom.py:9:0: E1120: No value for argument 'url' in function call (no-value-for-parameter)
```

I think the desired behavior would be to not raise this error.

### Command used

```shell
python3.9 -m pylint example.py --disable=all --enable=E
```

### Pylint output

```shell
************* Module phantom
example.py:9:0: E1120: No value for argument 'url' in function call (no-value-for-parameter)
```

### Expected behavior

No errors

### Pylint version

```shell
pylint 2.12.2
astroid 2.9.3
Python 3.9.9 (main, Nov 21 2021, 03:23:42) 
[Clang 13.0.0 (clang-1300.0.29.3)]
```

### OS / Environment

MacOS

**Comment by Pierre-Sassoulas** [comment]
Closing as I cannot reproduce on the main branch anymore.
